test/parallel/test-child-process-fork-detached.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const fork = require('child_process').fork;
+const fixtures = require('../common/fixtures');
+
+const nonPersistantNode = fork(
+  fixtures.path('parent-process-nonpersistent-fork.js'),
+  [],
+  { silent: true });
+
+let childId = -1;
+
+nonPersistantNode.stdout.on('data', function(data) {
+  childId = parseInt(data, 10);
+  nonPersistantNode.kill();
+});
+
+process.on('exit', function() {
+  assert.notStrictEqual(childId, -1);
+  // Killing the child process should not throw an error
+  process.kill(childId);
```

Comment:
This strikes me as having significant potential for leaving stray processes around on CI. Not necessarily blocking, but surely something to keep an eye on after this lands....